### PR TITLE
Implement post_prefix and post_suffix consistently for all clients 

### PR DESCRIPTION
**Pull request by wilddeej**
Includes tests to verify new/updated functionality.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .PHONY: test
 test: pep8
 	export PYTHONPATH=$(pwd)/src;
-	cd tests \
+	export MEDIA_DIR=/tmp && cd tests \
```
Comment (aurelg) [inline Makefile]: For the sake of consistency, shouldn't the `export` be on a separate line, just like the `PYTHONPATH` is?

**Comment by aurelg** [comment]
Thanks for this new PR!

I notice you often use `self._post_suffix if self._post_suffix else ''`. It makes sense, since it is initialized with `None` ([GenericClient.py:26](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L26)). An alternative could be to initialize it with `''`, so that its value does not have to be tested against `None` in each client. The result would be a simpler / more readable code. This `''` would be instantiated when the GenericClient class is parsed, and would be shared by all its instances as a class attribute, unless overridden by line [238](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L238).

(This applies to `_post_prefix` as well).

BTW, I noticed a possible issue with the declaration of `_tags`, `_tag_filter_opts` and  `_url_shortener_opts` at lines 18, 19 and 22 in [GenericClient.py](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L18-L19). This list and these two dicts are instantiated when parsing the file. It has no side effect, since they are assigned later a new value, if needed (lines [203](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L203), [209](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L209) and [244](https://github.com/wilddeej/feedspora/blob/enhancement/client-post-prefix-suffix/src/feedspora/generic_client.py#L244), respectively). However, this opens the door to possible mistakes later, just like [this one](https://github.com/aurelg/feedspora/pull/46#issuecomment-447628406). IMHO, just in case, they should be set to `None` instead.

**Comment by wilddeej** [comment]
Excellent points.  I've implemented all as suggested.

**Comment by wilddeej** [comment]
The change to _tags, _tag_filter_opts, and _url_shortener_opts have broken all of the tests.  I suspect this is because the `set_common_opts` function is not called by any of the tests, and so these variables can never get properly initialized.
I see four possibilities:
1. Change a lot of code to correctly respond to values of None for these variables,
2. Change all tests to call `set_common_opts`,
3. Introduce an __init__ function into the GenericClient where these values will be initialized,
4. Put the potentially dangerous class initialization back in place
Which do you think is best?

File: src/feedspora/generic_client.py [hunk 2]
```diff
@@ -199,12 +199,14 @@ def set_common_opts(self, account):
         '''
 
         # Tags
+        self._tags = []
```
Comment (aurelg) [inline src/feedspora/generic_client.py]: If `tags in account` is `True`, `self._tags` will be initialized by the list comprehension. If `tags in account` is `False`, then `self._tags` should be initialized with something compatible with its subsequent use in the code. The cleaner option would be not to change its default value (`None`) and handle it downstream in the code. However, an empty list is fine too. Its initialization would be clearer in an `else` statement, though.

File: src/feedspora/generic_client.py [hunk 3]
```diff
@@ -199,12 +199,14 @@ def set_common_opts(self, account):
...
         if 'tags' in account:
             self._tags = [
                 word.strip() for word in account['tags'].split(',')
             ]
 
         # Tag filtering options
+        self._tag_filter_opts = dict()
```
Comment (aurelg) [inline src/feedspora/generic_client.py]: Same here: the `_tag_filter_opts = dict()` should be in an `else`  statement.

File: src/feedspora/generic_client.py [hunk 4]
```diff
@@ -240,6 +242,7 @@ def set_common_opts(self, account):
         if 'url_shortener' in account:
             self._url_shortener = account['url_shortener'].lower()
 
+        self._url_shortener_opts = dict()
```
Comment (aurelg) [inline src/feedspora/generic_client.py]: Same here: the `_url_shortener_opts = dict()` should be in an `else`  statement.

File: src/feedspora/generic_client.py (hunk 2 above)
Comment (wilddeej) [reply src/feedspora/generic_client.py]: OK, but we need to work out the test functionality question below first, I think.

File: src/feedspora/generic_client.py (hunk 3 above)
Comment (wilddeej) [reply src/feedspora/generic_client.py]: OK, but we need to work out the test functionality question below first, I think.

File: src/feedspora/generic_client.py (hunk 4 above)
Comment (wilddeej) [reply src/feedspora/generic_client.py]: OK, but we need to work out the test functionality question below first, I think.

**Comment by aurelg** [comment]
>     1. Change a lot of code to correctly respond to values of `None` for these variables,

This would certainly be the cleaner option.

>     2. Change all tests to call `set_common_opts`,

If a call to `set_common_opts` is mandatory, then this call should be enforced at some point. I don't think calling it "manually" really makes sense.

>     3. Introduce an `__init__` function into the `GenericClient` where these values will be initialized,

This would be my second choice, and I can live with it if you don't want to update the code to deal with `None`.

>     4. Put the potentially dangerous class initialization back in place

I'd say that counting on something dangerous to solve a temporary problem is a dangerous bet. :-)

File: src/feedspora/generic_client.py (hunk 2 above)
Comment (aurelg) [reply src/feedspora/generic_client.py]: Indeed. I saw your comment after I posted mine.

**Comment by wilddeej** [comment]
All of the clients already call `set_common_opts` in their `__init__` functions, but that doesn't happen within tests, which have a test-specific initialization function injected.  Based on that, it doesn't sound like option 3 is viable, and wouldn't solve the problem in the tests anyway.
Both options 1 and 2 are really beyond the scope of this work, even though it might be the "right" thing to do.  I think it would be better to solve that in a more targeted effort, rather than trying to fit it in here.

**Comment by wilddeej** [comment]
I created Issue #49 for the client var cleanup issue, and this is fixed in PR #50 (including the "else" implementation of these variables as mentioned above)

**Pull request merged**